src/level.cpp

```diff
@@ -997,14 +998,250 @@ void Level::GenerateRasterizableData(std::vector<Quadblock>& quadblocks)
+		v.m_pos.z += radius; point(&v, 0, data); v.m_pos.z -= radius;
+
+		v.m_pos.x += radius; v.m_normal = Vec3(1.f / 1.44224957031f, -1.f / 1.44224957031f, 1.f / 1.44224957031f); point(&v, 0, data); v.m_pos.x -= radius;
+		v.m_pos.y -= radius; point(&v, 0, data); v.m_pos.y += radius;
+		v.m_pos.z += radius; point(&v, 0, data); v.m_pos.z -= radius;
+
+		v.m_pos.x += radius; v.m_normal = Vec3(1.f / 1.44224957031f, 1.f / 1.44224957031f, -1.f / 1.44224957031f); point(&v, 0, data); v.m_pos.x -= radius;
+		v.m_pos.y += radius; point(&v, 0, data); v.m_pos.y -= radius;
+		v.m_pos.z -= radius; point(&v, 0, data); v.m_pos.z += radius;
+
+		v.m_pos.x -= radius; v.m_normal = Vec3(-1.f / 1.44224957031f, -1.f / 1.44224957031f, 1.f / 1.44224957031f); point(&v, 0, data); v.m_pos.x += radius;
+		v.m_pos.y -= radius; point(&v, 0, data); v.m_pos.y += radius;
+		v.m_pos.z += radius; point(&v, 0, data); v.m_pos.z -= radius;
+
+		v.m_pos.x += radius; v.m_normal = Vec3(1.f / 1.44224957031f, -1.f / 1.44224957031f, -1.f / 1.44224957031f); point(&v, 0, data); v.m_pos.x -= radius;
+		v.m_pos.y -= radius; point(&v, 0, data); v.m_pos.y += radius;
+		v.m_pos.z -= radius; point(&v, 0, data); v.m_pos.z += radius;
+
+		v.m_pos.x -= radius; v.m_normal = Vec3(-1.f / 1.44224957031f, 1.f / 1.44224957031f, -1.f / 1.44224957031f); point(&v, 0, data); v.m_pos.x += radius;
+		v.m_pos.y += radius; point(&v, 0, data); v.m_pos.y -= radius;
+		v.m_pos.z -= radius; point(&v, 0, data); v.m_pos.z += radius;
+
+		v.m_pos.x -= radius; v.m_normal = Vec3(-1.f / 1.44224957031f, -1.f / 1.44224957031f, -1.f / 1.44224957031f); point(&v, 0, data); v.m_pos.x += radius;
+		v.m_pos.y -= radius; point(&v, 0, data); v.m_pos.y += radius;
+		v.m_pos.z -= radius; point(&v, 0, data); v.m_pos.z += radius;
+		};
+
+	/*for (Spawn& e : spawns)
+	{
+		Vertex v = Vertex(Point(e.pos.x, e.pos.y, e.pos.z, 0, 128, 255));
+		octohedralPoint(v, spawnsData);
+	}*/
+
+	for (Checkpoint& e : checkpoints)
+	{
+		Vertex v = Vertex(Point(e.Pos().x, e.Pos().y, e.Pos().z, 255, 0, 128));
+		octohedralPoint(v, checkData);
+	}
+
+	checkMesh.UpdateMesh(checkData, (Mesh::VBufDataType::VColor | Mesh::VBufDataType::Normals), Mesh::ShaderSettings::None);
+	this->m_checkModel.SetMesh(&checkMesh);
+}
+
+void Level::GenerateRenderStartpointData(std::array<Spawn, NUM_DRIVERS>& spawns)
+{
+	static Mesh spawnsMesh;
+	std::vector<float> spawnsData;
+
+	auto point = [](const Vertex* verts, int ind, std::vector<float>& data) {
+		//barycentricIndex is essentially "which index" is this vertex for the face.
+		data.push_back(verts[ind].m_pos.x);
+		data.push_back(verts[ind].m_pos.y);
+		data.push_back(verts[ind].m_pos.z);
+		data.push_back(verts[ind].m_normal.x);
+		data.push_back(verts[ind].m_normal.y);
+		data.push_back(verts[ind].m_normal.z);
+		Color col = verts[ind].GetColor(true);
+		data.push_back(col.r);
+		data.push_back(col.g);
+		data.push_back(col.b);
+		};
+
+	auto octohedralPoint = [&point](Vertex v, std::vector<float>& data) {
```

Comment:
aren't `point()` and `octohedralPoint()` duplicated in these two functions? They're good candidates to become private functions of the class